adds random quotes to the collection selection menu

bjsi opened the pull request:
Selects a random quote from a text file of ~300 Woz quotes. The file is assumed to be kept in the SMA root directory. Let me know if this is acceptable and I will add the file.

Example screenshot:

![image](https://user-images.githubusercontent.com/58147075/74280040-e3377000-4d13-11ea-9795-acb95d270eac.png)

bjsi commented:
quotes.csv is here: https://github.com/bjsi/Random-Woz-Quote/blob/master/quotes.csv

alexis- commented on src/AppHosts/SuperMemoAssistant/CollectionSelectionWindow.xaml.cs:
```diff
@@ -244,6 +246,36 @@ private void ListBoxItem_MouseDoubleClick(object sender, MouseButtonEventArgs e)
+
+      if (QuoteFile.Exists())
+      {
+        var Lines = File.ReadAllLines(QuoteFile.FullPath);
```
There are several scenarios where `File.ReadAllLines` could throw an exception, can you please:
- handle `FileIOException` with a `LogTo.Warning`,
- handle `Exception` (fallback) with a `LogTo.Error` (this will file a bug report as well as logging)
- let the try... catch encompass the whole if code branch (under the if), for safety
- gracefully continue the code if an exception is thrown (don't rethrow or exit)

alexis- commented on src/AppHosts/SuperMemoAssistant/CollectionSelectionWindow.xaml.cs:
```diff
@@ -244,6 +246,36 @@ private void ListBoxItem_MouseDoubleClick(object sender, MouseButtonEventArgs e)
+        // Field 1 = quote author
+        // Field 2 = quote URL
+        // Field 3 = quote title
+        if (!SplitQuoteLine[0].EndsWith(".")
```
This will cause an error if the file is empty

bjsi commented:
Ok @alexis- I added exception handling and a check for an empty file. I also added a hyperlink to the title of the quote. I changed the file to a .tsv because github can prettify .tsv files - maybe this can allow the community to add some of their own favourite Woz quotes or quotes from supermemo lore / the discord.

alexis- commented on src/AppHosts/SuperMemoAssistant/CollectionSelectionWindow.xaml.cs:
```diff
@@ -246,34 +249,59 @@ private void ListBoxItem_MouseDoubleClick(object sender, MouseButtonEventArgs e)
+          // Field 1: Author
+          // Field 2: Url
+          // Field 3: Title
+          if (!SplitQuoteLine[0].EndsWith(".")
```
Just nitpicking, but you could implement the condition this way to make the code more easy to read and maintain:
```
private const string[] SentenceEndingPunctuation = new[] { ".", "!", "?" };
[...]
if (SentenceEndingPunctuation[0].None(p => SplitLineQuoteLine.EndsWith(p)))
```

alexis- commented on src/AppHosts/SuperMemoAssistant/CollectionSelectionWindow.xaml.cs:
```diff
@@ -246,34 +249,59 @@ private void ListBoxItem_MouseDoubleClick(object sender, MouseButtonEventArgs e)
+          }
+
+          QuoteBodyTextBlock.Text = "\"" + SplitQuoteLine[0] + "\"";
+          QuoteAuthorTextBlock.Text = SplitQuoteLine[1];
```
That and subsequent array accesses [2] [3] will throw an exception if `QuoteLine` isn't correctly formatted (i.e. if it isn't a \t separated line as you expect).

It is very important to always assume your inputs are going to be somehow faulty (or even malicious), and to add the necessary code to deal with that.

alexis- commented on src/AppHosts/SuperMemoAssistant/CollectionSelectionWindow.xaml.cs:
```diff
@@ -246,34 +249,59 @@ private void ListBoxItem_MouseDoubleClick(object sender, MouseButtonEventArgs e)
+            SplitQuoteLine[0] += ".";
+          }
+
+          QuoteBodyTextBlock.Text = "\"" + SplitQuoteLine[0] + "\"";
```
FYI, there is a neat feature in C# called string interpolation. It's not very useful in this case, but often it will make it easier to read strings with embedded variables. Either:
- `$"\"{SplitQuoteLine[0]}\""`
- `$@"""{SplitQuoteLine[0]}"""` (double "" equals escaped ")

alexis- commented:
Can you merge upstream to resolve the possible conflict, and add the clause for the possible exception so we can add this feature to next version ?

bjsi commented:
Added suggestions here -> https://github.com/supermemo/SuperMemoAssistant/pull/142
